react/Menu/index.jsx

```diff
@@ -132,16 +162,91 @@ class Menu extends Component {
   }
 }
 
+class IternalMenuItem extends Component {
+  constructor() {
+    super()
+    this.element = ''
+    this.state = {
+      style: { display: 'block', visibility: 'hidden', position: 'fixed' }
+    }
+  }
+  //When the component is ready, than let's do some Maths
+  componentDidMount() {
+    this.handleDisplay()
+  }
+
+  handleDisplay() {
+    const { coordinates, itemsStyle, scrollTop } = this.props
+    //Find me & give me my coordinates
+    // eslint-disable-next-line
+    const domItem = ReactDOM.findDOMNode(this.element)
+    const domItemCoordinates = domItem.getBoundingClientRect()
+    //If we have don't have enough space => maths
+    const hasEnoughNotSpace =
+      domItemCoordinates.bottom - scrollTop > window.innerHeight
+    this.setState(
+      {
+        style: {
+          ...itemsStyle,
+          top: hasEnoughNotSpace
+            ? coordinates.top - domItemCoordinates.height
+            : coordinates.top + coordinates.height,
+          left: coordinates.left + coordinates.width - domItemCoordinates.width,
+          visibility: 'hidden',
+          position: 'fixed',
+          display: 'block'
+        }
+      },
+      () => {
+        /* We need callback here, because sometimes, getBoundingClient() 
+            doesn't return the right style... 
+          */
+        const updatedItem = domItem.getBoundingClientRect()
+
+        this.setState({
+          style: {
+            ...itemsStyle,
+            ...this.state.style,
+            top: hasEnoughNotSpace
+              ? coordinates.top - updatedItem.height
+              : coordinates.top + coordinates.height,
+            left: coordinates.left + coordinates.width - updatedItem.width,
+            visibility: 'visible'
+          }
+        })
+      }
+    )
+  }
+
+  render() {
+    return (
+      <div
+        className={cx(styles['c-menu__inner'], styles['c-menu__inner--opened'])}
+        ref={element => {
+          if (element !== null && element !== this.element) {
+            this.element = element
+          }
+        }}
+        style={({ ...this.props.itemsStyle }, { ...this.state.style })}
+      >
+        {this.props.renderItems()}
```

Comment:
Why not render the items in the parent and simply render `this.props.children` here ?

Reply:
It was done that way. I didn't want to change it to much. But will give a look.